eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelay.scala

```diff
@@ -89,6 +89,8 @@ object NodeRelay {
       Some(TrampolineFeeInsufficient)
     } else if (upstream.expiryIn - payloadOut.outgoingCltv < nodeParams.expiryDelta) {
       Some(TrampolineExpiryTooSoon)
+    } else if (CltvExpiry(nodeParams.currentBlockHeight) > payloadOut.outgoingCltv) {
+      Some(IncorrectOrUnknownPaymentDetails(upstream.adds.head.amountMsat, nodeParams.currentBlockHeight))
```

Comment:
I don't think this should be an `IncorrectOrUnknownPaymentDetails`, this is really a final recipient error for payments that don't match what's in the payment database...
Why didn't you use `TrampolineExpiryTooSoon`?

Reply:
Mainly because it has a chain height hint, but if this breaks the rules too much `TrampolineExpiryTooSoon` seems good enough.